week0003/truongngodang/java/Week0003.java

```diff
@@ -0,0 +1,128 @@
+import java.util.*;
+
+public class Week0003 {
+
+    public static void main(String[] args) {
+
+        Scanner scan = new Scanner(System.in);
+        ArrayList<String> stringRequest = new ArrayList<String>();
+        ArrayList<ProcessStringRequest> processRequests = new ArrayList<ProcessStringRequest>();
+        HashMap<String, Client> clients = new HashMap<String, Client>();
+        ArrayList<String> output = new ArrayList<String>();
+
+        for (int i = 0; scan.hasNext(); i++) {
+            stringRequest.add(scan.nextLine());
+            processRequests.add(new ProcessStringRequest(stringRequest.get(i)));
```

Comment:
If you have 1 million lines? You can't buffer like this. For every line, process it and move to the next line.